tests/tests/test_os_ent_migration.py

```diff
@@ -0,0 +1,240 @@
+
+    # create and accept some devs; save tokens
+    devs = []
+    for i in range(10):
+        devs.append(make_accepted_device(dauthd, dauthm, utoken))
+
+    # get tokens for all
+    for d in devs:
+        body, sighdr = deviceauth_v1.auth_req(
+                            d.id_data,
+                            d.authsets[0].pubkey,           
+                            d.authsets[0].privkey)
+
+        r = dauthd.call('POST',
+                        deviceauth_v1.URL_AUTH_REQS,
+                        body,
+                        headers=sighdr)
+
+        assert r.status_code == 200
+        d.token=r.text
+
+    return {"os_devs": devs, "os_users": users}
+
+def migrate_ent_setup():
+    """ Migrate the ENT setup - create a tenant and user via create-org,
+        substitute default token env in the ent. testing layer. 
+    """
+    ensure_conductor_ready(60, 'create_organization')
+
+    u = User('', 'baz@tenant.com', 'correcthorse')
+
+    cli = CliTenantadm(docker_prefix=docker_compose_instance)
+    tid = cli.create_org('tenant', u.name, u.pwd)
+    time.sleep(5)
+
+    tenant = cli.get_tenant(tid)
+
+    tenant = json.loads(tenant)
+    ttoken = tenant['tenant_token']
+
+    sed = "sed 's/$DEFAULT_TENANT_TOKEN/{}/' ".format(ttoken) + \
+          os.path.join(COMPOSE_FILES_PATH, "docker-compose.testing.enterprise.yml.template") + \
+          " > " + \
+          os.path.join(COMPOSE_FILES_PATH, "docker-compose.testing.enterprise.yml")
+
+    subprocess.check_call(sed, shell=True)
+
+    t = Tenant('tenant', tid, ttoken)
+    t.users.append(u)
+
+    return {"tenant": t}
+
+@pytest.mark.usefixtures("migrated_enterprise_setup")
+class TestEntMigration:
+    def test_users_ok(self, migrated_enterprise_setup):
+        uadmm = ApiClient('https://{}/api/management/v1/useradm'.format(get_mender_gateway()))
+
+        # os users can't log in
+        for u in migrated_enterprise_setup["os_users"]:
+            r = uadmm.call('POST', useradm.URL_LOGIN, auth=(u.name, u.pwd))
+            assert r.status_code == 401
+
+        # but enterprise user can 
+        ent_user = migrated_enterprise_setup["tenant"].users[0]
+        r = uadmm.call('POST', 
+                       useradm.URL_LOGIN, 
+                       auth=(ent_user.name, ent_user.pwd))
+        assert r.status_code == 200
+
+    def test_devs_ok(self, migrated_enterprise_setup):
+        uadmm = ApiClient('https://{}/api/management/v1/useradm'.format(get_mender_gateway()))
+        dauthd = ApiClient('https://{}/api/devices/v1/authentication'.format(get_mender_gateway()))
+        dauthm = ApiClient('https://{}/api/management/v2/devauth'.format(get_mender_gateway()))
+        depld = ApiClient('https://{}/api/devices/v1/deployments'.format(get_mender_gateway()))
+
+        # current dev tokens don't work right off the bat
+        # the deviceauth db is empty
+        for d in migrated_enterprise_setup["os_devs"]:
+            resp = depld.with_auth(d.token).call(
```

Comment:
Is this supposed to work afterwards, if we retry it at the end? Maybe we should?

Reply:
no, old tokens are dead forever.

we could accept devices at the end of the test, and then ask for updates - that should succeed.

but I stop at making sure devices are ready for acceptance - because the (later) accept + make sure devices can get auth pattern is tested all over.

Reply:
Yes, but is it tested after an upgrade? I reckon it's worth testing this here, because the consequences of this not working would be very bad (lose ability to deploy to devices after upgrading the server). And something could go wrong with the deployments database, for example.

If you still think it's not the proper place, I'm fine with that though. I'm not an expert in this area. :-)

Reply:
I'd leave it alone tbh. We have to stop somewhere, and there's a myriad of other things we could also test here

Reply:
Ok!